src/utils/string/index.ts

```diff
@@ -167,3 +167,27 @@ export const applyStrictRegex = (str: string) => `^${str}$`;
  * @return {string} A shortenend version of the input string.
  */
 export const shortenGithash = (str: string) => str?.substring(0, 7);
+
+/**
+ * Function that trims the middle portion of a string. ex: "EvergreenUI" -> "Ev...UI"
+ * If a string is longer than maxLength, it gets cropped exactly to maxLength.
+ * @param {string} text - Text to trim
+ * @param {number} maxLength - Max length before trimming text
+ * @param {number} trailingLength - Desired length of trailing text
+ * @return {string} The original or trimmed text.
+ */
+export const trimMiddleText = (
+  text: string,
+  maxLength: number,
+  trailingLength: number
+): string => {
+  const ellipsis = "...";
+  if (text.length > maxLength) {
+    return (
+      text.substring(0, maxLength - trailingLength - ellipsis.length) +
+      ellipsis +
+      text.substring(text.length - trailingLength)
```

Comment:
This still doesn't technically trim the middle of the string. It seems biased towards the head of the string.  (As seen in the test case below) 
I think ideally it would try to trim from the middle
`task_name -> ta...me` instead of `t...ame` 

It might also make sense to simplify the function a little bit so its easier to implement. 
One way we could do this 

```
const trimStringFromMiddle = (str, maxLength) => {
  if (str.length <= maxLength) {
    return str;
  }
  const midpoint = Math.floor(maxLength / 2);
  return `${str.substr(0, midpoint)}...${str.substr(str.length - midpoint)}`;
}
```

Reply:
Hmm yeah I guess I didn't name this function really accurately, it's more like `trimTextToFixedTrailingLength`... 🙂 I'll try again and write the true middle cropping function